Index: topicref.xsl
===================================================================
--- topicref.xsl	(revision 458)
+++ topicref.xsl	(working copy)
@@ -6,22 +6,19 @@
     <xsl:variable name="apos">'</xsl:variable>
     <xsl:variable name="jsapos">\'</xsl:variable>
     <xsl:variable name="comma">,</xsl:variable>
-    
-    <xsl:variable name="self">
-      <xsl:call-template name="stripstring">
-        <xsl:with-param name="jsvarstring" select="@navtitle"/>
-      </xsl:call-template>
-    </xsl:variable>
-    
+	<xsl:variable name="empty_string" select="''" />
+	<xsl:variable name="self" select="generate-id()"/>
+	
     <!--<xsl:variable name="self" select="translate(translate(translate(@navtitle, '/\^&amp;|\¬`*.-) (%$£!+=',
       ''), $apos, ''), $comma, '')"/>-->
     
     <!--<xsl:variable name="self" select="translate(translate(@href, '/', ''), '.', '')"/>-->
     <xsl:if test="not(@toc='no')">
       <xsl:if test="@href">
-        <xsl:message>
-          <xsl:value-of select="@href"/> Parent topic is <xsl:value-of select="$parent"/>
-        </xsl:message>
+		<!-- Send an error to the commandline if the navtitle is empty -->
+		<xsl:if test="@navtitle = $empty_string">
+			<xsl:message>WARNING: Topic <xsl:value-of select="@href"/>, has an empty navtitle!</xsl:message>
+		</xsl:if>
 
         <xsl:text>var </xsl:text>
         <xsl:value-of select="concat('obj', $self)"/>
